File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());
        int ans = 0 ;
        ArrayList<Integer> list = new ArrayList<>();



        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            list.add(Integer.parseInt(st.nextToken()));
        }

        Collections.sort(list);

        int m = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < m; i++) {
            int x = Integer.parseInt(st.nextToken());

            int left = 0;
            int right = n-1;


            while(left <= right){

                int mid = (right+left)/2;

                if(list.get(mid) == x){
                    ans = 1;
                    break;
                }else if(list.get(mid) > x){
                    right = mid - 1;
                }else if(list.get(mid) < x){
                    left = mid + 1;
                }
            }
            if(ans == 1){
                sb.append("1"+"\n");
            }else{
                sb.append("0"+"\n");
            }
            ans = 0;
        }

        System.out.println(sb);





    }
}

File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include<stdio.h>
#include<string.h>
#include<algorithm>
using namespace std;

int t [100001];
int main(){

    int n,m,found,ans;
    scanf("%d",&n);
    for (int i = 0; i < n; ++i) {
        scanf("%d",&t[i]);
    }

    sort(t,t+n);

    scanf("%d",&m);
    for (int i = 0; i < m; ++i) {
        scanf("%d",&found);
        int left = 0,right = n-1;
        ans = 0;
        while(left <= right){

            int mid = (left+right)/2;

            if(t[mid] == found){
                ans = 1;
                break;
            }else if(found < t[mid]){
                right = mid - 1;
            }else if (found > t[mid]){
                left = mid + 1;
            }

        }
        printf("%d\n",ans);

    }




    return 0;
}